### PR TITLE
[Bookmarks Manager & PowerBI] - Feat/bookmarks manager

### PR DESCRIPTION
**Pull request by jaysencpp**
# Description

Please include a summary of the change and which issue is fixed.
List any external dependencies that are required for this change.

Completes: [AB#209123](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/209123)
Completes: [AB#209114](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/209114)
Completes: [AB#209129](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/209129)
Completes: [AB#210106](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/210106)
Completes: [AB#210107](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/210107)
Completes: [AB#210108](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/210108)

## Checklist:

-   [ ] I have performed a self-review of my own code.
-   [ ] I have linked my DevOps task using the AB# tag.
-   [ ] My code is easy to read, and comments are added where needed.
-   [ ] My code is covered by tests.

### CODE REVIEW

File: src/Core/PowerBiViewer/src/Api/powerBiViewerApi.ts [hunk 1]
```diff
@@ -22,6 +22,7 @@ export function createPowerBiViewer({
         return {
             ...state,
             [viewerId]: {
+                groupe: [],
```
Comment (jaysencpp) [inline src/Core/PowerBiViewer/src/Api/powerBiViewerApi.ts:25]: see if this is needed

Comment (jaysencpp) [inline src/Core/PowerBiViewer/src/Api/powerBiViewerApi.ts:25]: see if this is needed

File: src/Core/PowerBiViewer/src/Types/State.ts [hunk 2]
```diff
@@ -7,6 +7,7 @@ export interface CoreViewState {
 export interface ViewState {
     title: string;
     shortName: string;
+    groupe: string | string[];
```
Comment (jaysencpp) [inline src/Core/PowerBiViewer/src/Types/State.ts:10]: see if needed.

Comment (jaysencpp) [inline src/Core/PowerBiViewer/src/Types/State.ts:10]: see if needed.

File: src/packages/BookmarksManager/src/hooks/useGetBookmarks.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { useQuery } from 'react-query';
+import { getBookmarks } from '..';
+import { BookmarkResponse } from '../types';
+
+export const useGetBookmarks = (appKey: string) => {
+    const { data, isFetching, error } = useQuery<BookmarkResponse[]>(['bookmarks', appKey], () =>
```
Comment (jaysencpp) [inline src/packages/BookmarksManager/src/hooks/useGetBookmarks.ts:6]: add abortsignal

File: src/packages/BookmarksManager/src/hooks/useBookmarks.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+/**
+ * Hook that handles API calls to bookmark service. Use together with your own functions to handle capturing and applying bookmarks.
+ */
+export const useBookmarks = <TPayload extends unknown = unknown>(): UseBookmarkReturn<TPayload> => {
+    const queryClient = useQueryClient();
+    const { mutate: saveMutation } = useMutation(saveBookmark, {
+        onSuccess: () => queryClient.invalidateQueries('bookmarks'),
+    });
+    const { mutateAsync: bookmarkApplyAsync } = useMutation(applyBookmark);
```
Comment (jaysencpp) [inline src/packages/BookmarksManager/src/hooks/useBookmarks.ts:36]: mutate

File: src/packages/BookmarksManager/src/hooks/useBookmarkEvent.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+ */
+export const useBookmarkEvents = ({ saveFn, applyFn }: BookmarkEventsArgs) => {
+    const ev = new EventHub();
+
+    useEffect(() => {
+        const evt = ev.registerListener(CustomEventActions.SAVE, saveFn);
+        return () => {
+            evt();
+        };
+    }, [saveFn]);
```
Comment (jaysencpp) [inline src/packages/BookmarksManager/src/hooks/useBookmarkEvent.ts:22]: remove from dep areray

File: src/packages/BookmarksManager/src/components/BookmarkDropdown/HeaderButton.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    :hover {
+        color: ${tokens.colors.text.static_icons__secondary.rgba};
+        background: ${tokens.colors.interactive.primary__hover_alt.rgba};
+
+        > svg > path {
+            fill: ${tokens.colors.text.static_icons__secondary.rgba};
+        }
+    }
+`;
+//TODO Extract this component to a package or core components (its implemented two different places)
```
Comment (jaysencpp) [inline src/packages/BookmarksManager/src/components/BookmarkDropdown/HeaderButton.tsx:42]: common component

File: src/packages/BookmarksManager/src/components/BookmarkDropdown/CreateNewBookmark.tsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+                <ClickableIcon
+                    name="add"
+                    onClick={(e) => {
+                        setIsCreatingNew(true);
+                        e.stopPropagation();
+                    }}
+                />
+            </CreateNewBookmarkWrapper>
+        );
+    } else {
```
Comment (jaysencpp) [inline src/packages/BookmarksManager/src/components/BookmarkDropdown/CreateNewBookmark.tsx:32]: else

File: src/packages/BookmarksManager/src/hooks/useGetBookmarks.ts (hunk 3 above)
Comment (jaysencpp) [inline src/packages/BookmarksManager/src/hooks/useGetBookmarks.ts:6]: isloading

File: src/Core/WorkSpace/src/Components/DataViewerHeader/Header.tsx [hunk 8]
```diff
@@ -4,8 +4,10 @@ import { tokens } from '@equinor/eds-tokens';
 import { useFilterApiContext } from '@equinor/filter';
 import { ClickableIcon, Icon } from '@equinor/lighthouse-components';
 import { StatusBar } from '@equinor/lighthouse-status-bar';
+import { isProduction } from '@equinor/portal-client';
 import { useMemo } from 'react';
 import { FilterFilled } from '../../../../../components/Icon/FilterIconFilled';
+import { BookmarkDropdown } from '../../../../../packages/BookmarksManager/src';
```
Comment (jaysencpp) [inline src/Core/WorkSpace/src/Components/DataViewerHeader/Header.tsx:10]: import

File: src/Core/PowerBiViewer/src/Utils/getReportByPage.ts [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+ * @return {*}  {(FusionPowerBiOptions | undefined)}
+ */
+export function getReportByPage(
+    selectedPage: Partial<Page>,
+    reports: FusionPowerBiOptions[]
+): FusionPowerBiOptions | undefined {
+    if (!selectedPage?.pageTitle || !selectedPage?.pageId) {
+        return undefined;
+    }
+    // return reports.find((r) => r.pages.includes(selectedPage));
```
Comment (jaysencpp) [inline src/Core/PowerBiViewer/src/Utils/getReportByPage.ts:15]: see if this is good enough now

File: src/Core/PowerBiViewer/src/PowerBiViewer.tsx [hunk 10]
```diff
@@ -54,10 +38,59 @@ export function PowerBiViewer(props: PowerBiViewerProps): JSX.Element {
         setIsFilterActive((s) => !s);
     }
 
+    const handleApplyingBookmark = async (
```
Comment (jaysencpp) [inline src/Core/PowerBiViewer/src/PowerBiViewer.tsx:41]: clean up

**Review by Gustav-Eikaas — APPROVED** [review]

**Pull request merged**